cmd/kueuectl/app/list/list_workload.go

```diff
@@ -216,13 +226,16 @@ func (o *WorkloadOptions) Run(ctx context.Context) error {
 	return printer.PrintObj(list, o.Out)
 }
 
-func (o *WorkloadOptions) filterList(list *v1beta1.WorkloadList) {
+func (o *WorkloadOptions) filterList(list *v1beta1.WorkloadList, apiResourceLists map[string]*metav1.APIResourceList) {
 	if len(list.Items) == 0 {
 		return
 	}
 	filteredItems := make([]v1beta1.Workload, 0, len(o.LocalQueueFilter))
 	for _, wl := range list.Items {
-		if o.filterByLocalQueue(&wl) && o.filterByClusterQueue(&wl) && o.filterByStatuses(&wl) {
+		if o.filterByLocalQueue(&wl) && o.filterByClusterQueue(&wl) && o.filterByStatuses(&wl) &&
+			o.filterByObjectRef(&wl, o.ForObject.Name, isNameEqual) &&
+			o.filterByResourceType(&wl, o.ForObject.Type, apiResourceLists) &&
+			o.filterByObjectRef(&wl, o.ForObject.APIGroup, isAPIGroupEqual) {
```

Comment:
I think it's easier to follow if we have only one filter func , or at least rename `filterByResourceType` to something like `filterByOwnerResourceType`.

Reply:
Refactored functions filterByObjectRef and filterByResourceType for readability